x-pack/plugins/task_manager/server/saved_objects/schemas/task.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+export const taskSchemaV1 = schema.object({
+  taskType: schema.string(),
+  scheduledAt: schema.maybe(schema.string()),
+  startedAt: schema.maybe(schema.nullable(schema.string())),
+  retryAt: schema.maybe(schema.nullable(schema.string())),
+  runAt: schema.maybe(schema.string()),
+  schedule: schema.maybe(
+    schema.object({
+      interval: schema.duration(),
+    })
+  ),
+  params: schema.recordOf(schema.string(), schema.any()),
+  state: schema.recordOf(schema.string(), schema.any()),
+  stateVersion: schema.maybe(schema.number()),
+  traceparent: schema.maybe(schema.string()),
+  user: schema.maybe(schema.string()),
+  scope: schema.maybe(schema.arrayOf(schema.string())),
+  ownerId: schema.nullable(schema.string()),
```

Comment:
`ownerId` is defined as optional in the TaskInstance schema, so maybe this should be `schema.maybe(schema.nullable(schema.string()))`?

Reply:
Or maybe this is based off the `ConcreteTaskInstance` definition?

Reply:
Good catch, made me realise that I should use `SerializedConcreteTaskInstance` as it is used in the TaskStore.schedule of TM.

Updated the schema accordingly.

Reply:
It looks like `SerializedConcreteTaskInstance` inherits an `id` field from `ConcreteTaskInstance` that is a string, however, I don't see the `id` field in the actual task document?

Reply:
`id` is not saved in the doc, omited here: https://github.com/elastic/kibana/blob/main/x-pack/plugins/task_manager/server/task_store.ts#L553
 
but used as doc id in ES here: https://github.com/elastic/kibana/blob/main/x-pack/plugins/task_manager/server/task_store.ts#L171